tensorflow/contrib/rnn/python/ops/core_rnn_cell_impl.py

```diff
@@ -866,11 +866,11 @@ def __call__(self, inputs, state, scope=None):
           # Default initializer for embeddings should have variance=1.
           sqrt3 = math.sqrt(3)  # Uniform(-sqrt(3), sqrt(3)) has variance=1.
           initializer = init_ops.random_uniform_initializer(-sqrt3, sqrt3)
-
-        if type(state) is tuple:
-          data_type = state[0].dtype
-        else:
-          data_type = state.dtype
+          
+        s = state
```

Comment:
that would work too, nest.flatten(state)[0] would probably do more work than this for loop but it takes fewer lines so I'm not too fussed